pkg/app/app.go

```diff
@@ -55,11 +63,11 @@ type App struct {
 
 func NewApp(app v1alpha1.App, hooks Hooks,
 	fetchFactory fetch.Factory, templateFactory template.Factory,
-	deployFactory deploy.Factory, log logr.Logger, appMetrics *metrics.AppMetrics, compInfo ComponentInfo) *App {
+	deployFactory deploy.Factory, log logr.Logger, opts AppOpts, appMetrics *metrics.AppMetrics, compInfo ComponentInfo) *App {
```

Comment:
this reminds me of last time this function's arguments list got too long and i asked you how many is too many

Reply:
im ok with the current set since that is indeed all the dependencies App requires. may be in a separate PR id like to see if we can merge CRDApp and App together. that might simplify some of this.